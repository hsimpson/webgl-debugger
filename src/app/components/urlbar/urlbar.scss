.UrlBar {
  display: flex;
  flex-direction: row;
}
