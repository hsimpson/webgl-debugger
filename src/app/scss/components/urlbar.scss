.urlbar {
  display: flex;

  #urlinput {
    flex-grow: 1;
  }
}
