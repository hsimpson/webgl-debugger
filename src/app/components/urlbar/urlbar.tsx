import './urlbar.scss';
import React from 'react';
import { TextField } from '@material-ui/core';

/*
interface IUrlBarState {
  url: string;
}
*/

interface IUrlBarProps {
  url: string;
  onChange: (url: string) => void;
}

export class UrlBar extends React.Component<IUrlBarProps> {
  /*
  public readonly state: IUrlBarState = {
    url: '',
  };
  */

  private handleUrlChange = (event: React.ChangeEvent<{ name?: string; value: unknown }>): void => {
    this.props.onChange(event.target.value as string);
  };

  public render(): React.ReactNode {
    return (
      <div className="UrlBar">
        <TextField
          fullWidth
          margin="normal"
          id="outlined-basic"
          //className={classes.textField}
          label="Launch URL"
          variant="outlined"
          value={this.props.url}
          onChange={this.handleUrlChange}
        />
      </div>
    );
  }
}
