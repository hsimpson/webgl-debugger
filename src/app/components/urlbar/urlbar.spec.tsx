import { mount } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import * as TestSubject from './urlbar';

describe('UrlBar', (): void => {
  const url = 'http://ende.de';
  const onChange = jest.fn();

  beforeEach(() => {
    onChange.mockReset();
  });

  it('exists', (): void => {
    expect(TestSubject.UrlBar).toBeDefined();
  });

  it('snapshot', (): void => {
    const snapshot: renderer.ReactTestRenderer = renderer.create(<TestSubject.UrlBar url={url} onChange={onChange} />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('label', (): void => {
    const component = mount(<TestSubject.UrlBar url={url} onChange={onChange} />);
    const label = component.find('label');
    expect(label.text()).toBe('Launch URL');
  });

  it('onChange', (): void => {
    const component = mount(<TestSubject.UrlBar url={url} onChange={onChange} />);
    const input = component.find('input');
    expect(input.getDOMNode().getAttribute('value')).toBe(url);
    input.simulate('change');
    expect(onChange).toHaveBeenNthCalledWith(1, url);
  });
});
