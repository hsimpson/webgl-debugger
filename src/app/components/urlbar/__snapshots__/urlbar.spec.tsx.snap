// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlBar snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="outlined-basic"
      id="outlined-basic-label"
    >
      Launch URL
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="http://ende.de"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-61 PrivateNotchedOutline-legendNotched-62"
        >
          <span>
            Launch URL
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
